### PR TITLE
Enforce that the node name in ZK is same as snapshotId.

### PR DESCRIPTION
**Pull request by mansu**
Currently, the node name can be different from snapshotId but this flexibility introduces bugs that are really hard to debug. Since we don't need this flexibility for now, enforce this restriction by modifying the constructor.

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadata.java [hunk 1]
```diff
@@ -29,7 +52,7 @@ public SnapshotMetadata(
     checkState(startTimeUtc > 0, "start time should be greater than zero.");
     checkState(endTimeUtc > 0, "end time should be greater than zero.");
     checkState(endTimeUtc > startTimeUtc, "start time should be greater than  end time.");
-    checkState(maxOffset > 0, "max offset should be greater than zero.");
+    checkState(maxOffset >= 0, "max offset should be greater than zero.");
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadata.java:55]: We need to update the error message as well

File: kaldb/src/main/java/com/slack/kaldb/util/SnapshotUtil.java [hunk 2]
```diff
@@ -4,6 +4,6 @@
 
 public class SnapshotUtil {
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/util/SnapshotUtil.java:5]: This should be moved to the test package, as this is only for use in tests and not production code.

**Review by bryanlb — CHANGES_REQUESTED** [review]

File: kaldb/src/main/java/com/slack/kaldb/util/SnapshotUtil.java (hunk 2 above)
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/util/SnapshotUtil.java:5]: Good call. Moved.

File: kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadata.java (hunk 1 above)
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/metadata/snapshot/SnapshotMetadata.java]: Done.

**Review by bryanlb — APPROVED** [review]

**Pull request merged**